Update Protocol/Dummy.cpp

Phatcat opened the pull request:
Satisfy Clang/libc++

Will corrupt the library if an empty string (file) is passed:

```
[ 84%] Linking CXX executable gateway
ld: archive member '/' not a mach-o file in '/Users/holsthein/Ember/build/src/libs/protocol/protocol.a'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [src/gateway/CMakeFiles/gateway.dir/build.make:117: src/gateway/gateway] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1512: src/gateway/CMakeFiles/gateway.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```

Chaosvex commented:
I'll accept this for now as a workaround but it should be possible to remove the Dummy.cpp entirely.

Phatcat committed:
Update protocol/Dummy.cpp

Satisfy Clang/libc++
Will corrupt the library if an empty string (file) is passed.

Get rid of dummy.cpp and run it as a headers-only interface
